#!/usr/bin/env python

PACKAGE="daniels"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("scanResolution", int_t ,0,"Entf-Faktor 1=Aufloesung 1m. 100=1cm 50=2cm usw",100,1,1000)

gen.add("minBallRadius",int_t,0,"min Ball Radius in mm",135,10,1000)
gen.add("maxBallRadius",int_t,0,"max Ball Radius in mm",135,10,1000)
gen.add("stepBallRadius",int_t,0,"Step Ball Radius",10,1,100)

gen.add("houghCircleCount",int_t,0,"Anzahl Kreisepunkte pro LaserPunkt",180,1,720)

gen.add("minBallPeak",int_t,0,"Min Peak fuer BallErkennung",4000,1,20000)

gen.add("publishHough",bool_t,0,"Ob HoughImage published werden soll",False)
gen.add("useCostmap",bool_t,0,"Ob die Costmap zum Filtern verwendet werden soll",True)

gen.add("processEveryScan",int_t,0,"Jeden xten Scan verarbeiten",5,1,150)

exit(gen.generate(PACKAGE, "daniels", "ScanBall"))

